### PR TITLE
chore: cap show more to 6, and use for all long lists

### PR DESCRIPTION
**Pull request by mzikherman**
Closes https://artsyproduct.atlassian.net/secure/RapidBoard.jspa?rapidView=4&projectKey=FX&modal=detail&selectedIssue=FX-2704

The commits tell the story. Capped 'show more' to 6, and added it to the artists filter (on fair page), and the medium and time period filters, as well as tweaking the color filter from 7 -> 6.

Also generalized the somewhat simple autosuggest widget and added it to artist nationality and artwork location (in addition to partners).

### CODE REVIEW

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ArtistsFilter.tsx [hunk 1]
```diff
@@ -1,28 +1,69 @@
...
+    (isFollowedArtistCheckboxSelected && isFollowedArtist)
+  const props = {
+    onSelect: selected => {
+      toggleArtistSelection(selected, slug)
+    },
+    selected,
+  }
+
+  return (
+    <Checkbox {...props}>
```
Comment (dzucconi) [inline src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ArtistsFilter.tsx:61]: I will keep commenting on these, every time. ***Use JSX***

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/MediumFilter.tsx [hunk 2]
```diff
@@ -30,23 +31,25 @@ export const MediumFilter: FC = () => {
...
+            const selected =
+              currentFilters.additionalGeneIDs.includes(slug) ||
+              currentFilters.medium === slug
+            const props = {
+              key: index,
+              onSelect: selected => {
+                toggleMediumSelection(selected, slug)
+              },
+              selected,
+            }
```
Comment (dzucconi) [inline src/v2/Components/v2/ArtworkFilter/ArtworkFilters/MediumFilter.tsx:45]: Just give me a single decent reason :)

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ArtistsFilter.tsx (hunk 1 above)
Comment (mzikherman) [reply src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ArtistsFilter.tsx:61]: I moved this verbatim from elsewhere in this file, (altho I likely introduced the non-JSX props style originally 😅 )

**Review by dzucconi — APPROVED** [review]
Looks good. Glad these will be consistent.

Stylistically I really wish there was a good reason for the indirection of loading props into an object then spreading them rather than just doing it directly. It's just noise?

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/MediumFilter.tsx (hunk 2 above)
Comment (mzikherman) [reply src/v2/Components/v2/ArtworkFilter/ArtworkFilters/MediumFilter.tsx:45]: I think it's just personal pref. Happy to change it.

personally, my eyes are more drawn to 'code', in terms of variable assignments (`const checkbox, props, ...` etc.).

Then, the JSX fragment is always very 'clean' and I can just sort of glance through that.

vs. having a large/complex JSX fragment that sets lots of props, in addition to iteration and all that. I even like taking the `map` stuff out and moving those to a separate `renderGroup(..)` call to clean up the top level JSX even more (but i've stopped that in this PR after being called out earlier).

Comment (dzucconi) [reply src/v2/Components/v2/ArtworkFilter/ArtworkFilters/MediumFilter.tsx:45]: It just defeats the purpose of JSX and is unconventional — like why not just go all the way and write direct `React.createElement` calls (don't do it).

It's not "clean" so much as it's indirect.

The render group stuff kills me too though — whenever I see something like that it's clear to me that the boundaries of the component could be improved and someone just doesn't want to make a new file or whatever.

**Comment by mzikherman** [comment]
Updated to move away from separating props assignments from JSX.

**Pull request merged**